### DISCUSSION TITLE
Curseforge launcher keeps on pushing fabric version even though I use forge

### DISCUSSION THREAD

**Discussion by Rychlas** [RakambdaOrg/FallingTree#74] [Q&A] [answered]
Greetings, I use Forge, 1.16.5-2.10.2 version of the mod.
I'm not sure whether I'm doing something wrong but whenever I press "update" on the curseforge mod manager to update the mod, it installs a fabric version instead. I just checked, and even though I have the most recent forge version, I can see the update button, and when I press it it removes the Forge version, installs the Fabric one along with its prerequisite mod lib. 

Hence, the question - is it a curseforge mod manager quirk, did I do something wrong and I need to select the forge version somehow somewhere, or is there some kind of setting to be set from the author of the mod's side? 

If it is entirely my fault, I apologise for wasting your time. 
Kudos to you for a great mod.

**Comment by Rakambda** [comment]
I don't really know what is the `curseforge mod manager` that you use, but I guess you refer to the Curse client on desktop. If so then this is an "issue" on their side.

I put issue between quotes because I don't really know if it is an issue or a choice they had. Basically as mod publishers we can tag the mod with mainly 2 different kind of tags: the minecraft version, and the mod loader.
* As you can see [here](https://www.curseforge.com/minecraft/mc-mods/falling-tree/files/3207457) it is tagged as `1.16.5` and for `Forge`.
* And [here](https://www.curseforge.com/minecraft/mc-mods/falling-tree/files/3207464) `1.16.5` for `Fabric`.

These tags are used when using the search onto the curse website, or when listing the files for a mod (eg: [getting only forge versions of the mod](https://www.curseforge.com/minecraft/mc-mods/falling-tree/files/all?filter-status=1&filter-game-version=2020709689:7498)).
The issue now comes to the desktop client, it doesn't care at all about that `Loader` tag. It just takes into account the version. So as my mod supports both Fabric & Forge, it'll basically get the last file that has been published for a given version (whatever the loader).

As you can see from the files page (https://www.curseforge.com/minecraft/mc-mods/falling-tree/files), I tend to first release the Forge version, and then the Fabric version. So from the desktop client point of view, the Fabric version is slightly newer than the Forge one.

---

Now there could be several alternatives to that:
* Manage your mods manually. Though I agree it isn't very appealing.
* As an author I could split this mod into 2 distinct projects on curse (`FallingTree (Forge)` & `FallingTree (Fabric)`).
* Make Curse desktop client change somehow.
* Use other mod loaders

For the first point, I don't really consider it a good option as most people don't wanna handle that manually nor have enough knowledge to do it in a comfortable way.

For the second point, this has already been suggested to me. But for now it'll stay as it is. I get the point of view using the desktop client, but to me it should be the 3rd point that is done. They allow us creators to tag files with the modloader, so why not taking it into account in the desktop client?

For the last point, I don't really know if there is any that exists. But there's an API available to retrieve files/tags and so it should be doable to have a "community made" mod loader that can do what the curse desktop does but taking into account the loader.

---

TL:DR: Curse desktop client doesn't care about the modloader so it takes the most recent file for a given MC version. And also I'm somewhat stubborn and am not planning in the near future to split the releases into two projects on curseforge.

**Reply by Rychlas** [reply]
Oh, sure, thank you for the explanation. I'll just try to remember not to hit the update button by accident, it's not a huge issue. I was just wondering whether I made a mistake somewhere. And I agree with you that they should be more consistent with how they code their things. Especially since both website and the Curse loader are theirs, it's weird they do not use each other's features properly.

**Comment by Rakambda** [comment]
An update to what has been said previously:

I recently tried to ship the mods for Forge and Fabric inside the same Jar file (#90). This means that there isn't any distinction anymore between Forge and Fabric (only internally in the mod).
I guess this should provide a sort of better solution to the "curseforge should handle it better". This way every update you pull from the launcher should be compatible with Forge & Fabric.

This starts with version 2.11.1 and will be carried for next versions.
https://www.curseforge.com/minecraft/mc-mods/falling-tree/files/3275003

**Comment by Rakambda** [comment, marked as answer]
Another update to this, recently curse made changes to allow mods to be tagged with several modloaders.

![image](https://user-images.githubusercontent.com/1688389/137755322-f30c4392-26aa-4762-935d-09795cf60c37.png)
https://discord.com/channels/428228256236306434/525372338653626389/899479301882269736

Starting with the next versions published (so probably 1.18+) they'll be tagged as so and therefore the launcher should handle them correctly.

**Reply by Rychlas** [reply]
Sweet, this pretty much solves everything. Thanks for updating!
